src/treelearner/serial_tree_learner.cpp

```diff
@@ -66,6 +72,48 @@ void SerialTreeLearner::Init(const Dataset* train_data, bool is_constant_hessian
     cegb_.reset(new CostEfficientGradientBoosting(this));
     cegb_->Init();
   }
+
+  if (config_->linear_tree) {
+    InitLinear(train_data, config_->num_leaves);
+  }
+}
+
+void SerialTreeLearner::InitLinear(const Dataset* train_data, const int max_leaves) {
```

Comment:
@btrotta I'm very inexperienced with C++, could you help me understand how the R package tests are working? Shouldn't compilation here fail if `Eigen/Dense`  isn't included? And won't it not be included for the R package because of `ifndef LGB_R_BUILD`?

Reply:
The only code that uses `Eigen/Dense` is the method `CalculateLinear`. This method only gets called if `linear_tree=True`. If `LGB_R_BUILD` is defined, then the code at line 878 (https://github.com/microsoft/LightGBM/pull/3299/files/4410c9cd1313ffc7333ca077e488296baac6f0ea#diff-110aae7c40f3150eb5d8264c1cb0663eR878) means that the compiler will compile a different version of `CalculateLinear` that just throws an error if it gets called (which would happen if an R-package user tried to set `linear_tree=True`).